streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -900,6 +902,14 @@ private long pollPhase() {
 
         final int numRecords = records.count();
 
+        for (final TopicPartition topicPartition: records.partitions()) {
+            records
+                .records(topicPartition)
+                .stream()
+                .max(Comparator.comparing(ConsumerRecord::offset))
+                .ifPresent(t -> taskManager.updateTaskEndMetadata(topicPartition, t.offset()));
```

Comment:
Dumb question: what do the TaskMetadata's `endOffsets` actually refer to? The KIP says that they reflect `"the highest offset seen so far"`, but I'm wondering what "seen" means in this context: is it the highest offset processed, or the highest offset consumed? I'm guessing the former since that feels more useful, but maybe an example of how this will be used will help me understand. (Asking because if it is the former, then we don't want to update them based on records that have been polled but not processed)

Also, can you add javadocs explaining this and the other TaskMetadata getter methods like you had in the KIP?

Reply:
It was intended to be the highest offset that the client knows is in the topic. In order to make that the case I am looking the most recent pull now.

I will add the java docs to the TaskMetadata getters